src/dmd/e2ir.d

```diff
@@ -1945,7 +1945,7 @@ elem *toElem(Expression e, IRState *irs)
                     elem *earray = ExpressionsToStaticArray(ne.loc, ne.arguments, &sdata);
 
                     e = el_pair(TYdarray, el_long(TYsize_t, ne.arguments.dim), el_ptr(sdata));
-                    if (irs.params.targetOS == TargetOS.Windows && irs.params.is64bit)
+                    if (target.os == Target.OS.Windows && irs.params.is64bit)
```

Comment:
More broadly speaking, these conditions should be replaced by a target method/hook for better cross compilation support.

For instance this cryptic condition and other similar ones for calling addressElem seem to be for determining whether the target passes dynamic arrays by reference. So they could all potentially be replaced with `target.passDArrayByRef` or `target.passByReference(ty)`.

Once all have been converted, OS and CPU information can just be made private.